nodeJS/express/views.md

```diff
@@ -0,0 +1,178 @@
+
+In this course, we will use [EJS](https://ejs.co/). EJS's syntax is very similar to that of HTML, meaning that the learning curve is relatively low compared to other template engines.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- How to setup EJS in an Express project
+- How to use EJS 
+
+### Setting up EJS
+
+Let's get started with EJS! Install EJS into your application by typing the following into your terminal:
+
+```bash
+npm install ejs
+```
+
+At the root of your project, create a subfolder called `views`.
+
+Next, we need to let our app know that we intend to use `EJS` as a template engine, as well as where to look for view files.
+
+In your `app.js` file, set the following application properties:
+
+```javascript
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "ejs");
+```
+
+This enables EJS as the view engine, and that our app should look for templates in the `/views` subdirectory.
+
+### EJS syntax
+
+In EJS, the `<%` and `%>` tags allow us to use JavaScript. This lets us write conditional statements, `for` loops, as well as use variables.
+
+In order to output a variable as a value, we use the `<%=` tag. 
+
+Here's a quick example that makes includes arrays and loop logic.
+
+```javascript
+<% const animals = ["Cat", "Dog", "Lemur", "Hawk"] %>
+
+<% animals.map(animal => { %>
+- <%= animal%>s are cute <% }) %>
+```
+
+### Using EJS with Express
+
+Let's use EJS with Express. First, create an EJS template file called `index.ejs` in the `views` subdirectory, and add the following:
+
+```html
+<html>
+  <body>
+    <%= message %>
+  </body>
+</html>
+```
+
+And in app.js, render this template file in one of your routes:
+
+```javascript
+app.get("/", (req, res) => {
+  res.render("index", { message: "EJS rocks!" });
+});
+```
+
+Start the server and go to the `/` route in the browser. You should see:
+
+```html
+EJS rocks!
+~~~
+
+If you inspect the HTML in the browser's dev tools, you can see HTML is structures exactly like how we wrote the EJS template with the `message` variable replaced with its value.
+
+When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });`  is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a tempalte called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template.
```

Comment:
```suggestion
When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });`  is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template.
```